src/components/ContentWidgets/SideBar/SideBar.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from 'react'
+import { ContentViewProps } from '../../ContentView/ContentView';
+import { ContentData } from '../../_data/Data';
+
+export type SideBarProps = {
+    contentData: ContentData,
+    pageTitle: string,
+    activeSection: string,
+    activeSectionIndex: number
+}
+
+export const SideBar: React.FC<SideBarProps> = ({
+    contentData,
+    pageTitle, 
+    activeSection,
+    activeSectionIndex
+}) => {
+
+    if (!contentData && !pageTitle) {
+        return <></>;
+    } else if (!contentData[pageTitle].hasSidebar) {
+        return <></>;
+    }
+
+    const onClick = (event: React.MouseEvent) => {
+        let location = document.getElementById(event.currentTarget.id.split("-")[0])!!.getBoundingClientRect().top - 100;
+        window.scrollTo({ top: location, behavior: 'smooth' });
+    }
+console.log(activeSectionIndex)
+    let progressStyle = {
+        height: activeSectionIndex + "%"
+    };
+    return <div id="sidebar-container">
+        <ul id="sidebar">
+            {contentData[pageTitle].content && contentData[pageTitle].content ? 
+                contentData[pageTitle].contentOrder!!.map((content, index) => {
+                    if (contentData[pageTitle].content!![content]!!.type == "HEADER") {
```

Comment:
Factor out variables used multiple times into one variable (`contentData[pageTitle.content!![content]!!`)

and also do explicit null checks instead of double bang assertion.